telepict/static/js/canvas-draw.ts

```diff
@@ -0,0 +1,258 @@
+      this.w = Math.round(wrapper.offsetWidth * this.pixelRatio);
+      this.h = Math.round(wrapper.offsetHeight * this.pixelRatio);
+    }
+    if (!wrapper) wrapper = document.createElement('div');
+    wrapper.innerHTML = '';
+
+    const styleWidth = `${Math.round(this.w / this.pixelRatio)}px`;
+    const styleHeight = `${Math.round(this.h / this.pixelRatio)}px`;
+
+    this.wrapper = h(wrapper, undefined, [
+      h('div', {className: 'top-controls', style: {marginBottom: '4px'}}, [
+        h<HTMLButtonElement>('button', {name: 'undo', onclick: this.undo}, ["Undo"]), " ",
+        h<HTMLButtonElement>('button', {name: 'clear', onclick: this.clear}, ["Clear"]),
+      ]),
+      h('div', {className: 'left-controls', style: {marginBottom: '4px'}}, [
+        h<HTMLButtonElement>('button', {onclick: this.clickColor, value: "black"}, [
+          h('span', {className: 'color', style: {background: 'black', display: 'inline-block', width: '12px', height: '12px'}}),
+        ]), " ",
+        h<HTMLButtonElement>('button', {onclick: this.clickColor, value: "#F55252"}, [ // red
+          h('span', {className: 'color', style: {background: '#F55252', display: 'inline-block', width: '12px', height: '12px'}}),
+        ]), " ",
+        h<HTMLButtonElement>('button', {onclick: this.clickColor, value: "#F8BC01"}, [ // yellow
+          h('span', {className: 'color', style: {background: '#F8BC01', display: 'inline-block', width: '12px', height: '12px'}}),
+        ]), " ",
+        h<HTMLButtonElement>('button', {onclick: this.clickColor, value: "#3DC853"}, [ // green
+          h('span', {className: 'color', style: {background: '#3DC853', display: 'inline-block', width: '12px', height: '12px'}}),
+        ]), " ",
+        h<HTMLButtonElement>('button', {onclick: this.clickColor, value: "#42B0FF"}, [ // blue
+          h('span', {className: 'color', style: {background: '#42B0FF', display: 'inline-block', width: '12px', height: '12px'}}),
+        ]), " ",
+        h<HTMLButtonElement>('button', {onclick: this.clickColor, value: "#D512F9"}, [ // purple
+          h('span', {className: 'color', style: {background: '#D512F9', display: 'inline-block', width: '12px', height: '12px'}}),
+        ]), " ",
+        h<HTMLButtonElement>('button', {onclick: this.clickColor, value: "#8D6E63"}, [ // brown
+          h('span', {className: 'color', style: {background: '#8D6E63', display: 'inline-block', width: '12px', height: '12px'}}),
+        ]),
+        " | ",
+        h<HTMLButtonElement>('button', {onclick: this.clickStrokeWidth, value: "1"}, [
+          "Thin",
+        ]), " ",
+        h<HTMLButtonElement>('button', {onclick: this.clickStrokeWidth, value: "2"}, [
+          "Normal",
+        ]), " ",
+        h<HTMLButtonElement>('button', {onclick: this.clickStrokeWidth, value: "4"}, [
+          "Thick",
+        ]), " ",
+      ]),
+      h('div', {style: {width: styleWidth, height: styleHeight, border: `1px solid gray`}}, [
```

Comment:
@dpitch40 It already has a border, it's here. Is it not showing up? Do you have a screenshot?

Reply:
![CanvasDraw](https://user-images.githubusercontent.com/1520538/86522006-49d13d80-be1d-11ea-868e-f51bce2c80f1.png)

I see one on the top edge and presumably the left, but not the bottom or right.

Reply:
LOL, your CSS (Bootstrap) smashes box sizing. That's why it was cutting off the border.

```css
*,
*::before,
*::after {
  box-sizing: border-box; // 1
}
```

Here, this will fix the border.

```suggestion
      h('div', {style: {width: styleWidth, height: styleHeight, border: `1px solid gray`, boxSizing: 'content-box'}}, [
```